### PR TITLE
chore: upgrade docs-kit to v21

### PR DESCRIPTION
**Pull request by emmenko**
* Reverts the MDX v2 upgrade (from docs-kit v20)
* Upgrade Gatsby to v5
* Set Node.js version to v18

### CODE REVIEW

**Review by nkuehn — APPROVED** [review]
Thanks @emmenko  

I took the update the actual config directly here so we don't have to do another iteration 
 - GA and hubspot IDs
 - updated beta link
 - updated flags to a combination that makes sense, updated link to flags docs

**Review by Rhotimee — APPROVED** [review]

**Comment by emmenko** [comment]
@nkuehn Lovely, thanks!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -88,7 +88,7 @@ jobs:
         uses: ./.github/actions/ci
 
       - name: Testing Starter application development server
-        run: yarn start-server-and-test 'yarn template-starter:start' 3001 'exit 0'
+        run: yarn start-server-and-test 'yarn template-starter:start' http-get://127.0.0.1:3001 'exit 0'
```
Comment (CarlosCortizasCT) [inline .github/workflows/main.yml:91]: Just curious about where this `http-get` syntax comes from.

**Review by CarlosCortizasCT — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (nkuehn) [reply .github/workflows/main.yml:91]: It's a special notation of the "start-server-and-test" library https://github.com/bahmutov/start-server-and-test#use-http-get-requests

Comment (CarlosCortizasCT) [reply .github/workflows/main.yml:91]: Thanks for the explanation 👍

**Pull request merged**